compiler/damlc/tests/daml-test-files/UnusedMatchTests.daml

```diff
@@ -1,13 +1,20 @@
 -- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its
 -- affiliates. All rights reserved.
 
--- @WARN range=12:2-12:3; Defined but not used
+-- @WARN range=19:2-19:3; Defined but not used
 
 {-# OPTIONS_GHC -Wunused-matches #-}
+{-# OPTIONS_GHC -Wunused-foralls #-}
+{-# OPTIONS_GHC -Wunused-imports #-}
+{-# OPTIONS_GHC -Wunused-pattern-binds #-}
+{-# OPTIONS_GHC -Wunused-top-binds #-}
+{-# OPTIONS_GHC -Wunused-type-patterns #-}
+-- We know this will fail and why.
+--   {-# OPTIONS_GHC -Wunused-local-binds #-}
```

Comment:
Can we enable this and add an expected diagnostics to make sure we only hit it in the one places we expect?

Reply:
Ok, let's see how that plays out!

Reply:
It's kind of a pain. As phrased we'll have to put in 48 `@WARN` lines. I think it hinders rather than helps. You ok with leaving it as is?

Reply:
Yeah if it’s that much definitely leave it out for now.

Reply:
Gracias muchos.